backend/prisma/schema.prisma

```diff
@@ -8,29 +8,34 @@ datasource db {
 }
 
 model User {
-  id               Int           @id @default(autoincrement())
-  username         String        @unique
-  email            String?       @unique
-  password         String?
-  fullName         String
-  imageUrl         String?
-  createdAt        DateTime      @default(now())
-  updatedAt        DateTime      @default(now()) @updatedAt
-  codeGroups       CodeGroup[]   @relation("UserCodeGroups")
-  groupMemberships GroupMember[] @relation("UserGroupMemberships")
-  addedGroups      GroupMember[] @relation("AddedByUser")
-  files            File[]        @relation("UserFiles")
+  id              Int           @id @default(autoincrement())
+  username        String        @unique
+  email           String?       @unique
+  password        String?
+  fullName        String
+  imageUrl        String?
+  availability    String?
+  createdAt       DateTime      @default(now())
+  updatedAt       DateTime      @default(now()) @updatedAt
+  skills          Skill[]
+  interests       Interest[]
+  groups_created  CodeGroup[]   @relation("GroupsCreated")
+  groups_added_to GroupMember[] @relation("GroupsAddedTo")
+  files_created   File[]        @relation("UserFiles")
 }
 
 model CodeGroup {
-  id        Int           @id @default(autoincrement())
-  groupName String
-  imgUrl    String?
-  creatorId Int
-  createdAt DateTime      @default(now())
-  creator   User          @relation("UserCodeGroups", fields: [creatorId], references: [id])
-  members   GroupMember[] @relation("GroupMembers")
-  files     File[]        @relation("GroupFiles")
+  id                     Int           @id @default(autoincrement())
+  groupName              String
+  imgUrl                 String?
+  preferred_category     String?
```

Comment:
This field is referred to as category here, but interests above. If these represent the same thing you should use consistent naming.

Reply:
@azhiah Yes, it does. I thought it made sense semantically, since its not the group's interest per se, It is a set project category.

Reply:
Ok, why is the field a string here but Interest type is used for the User field? Also a nit, but if it's a set category for the group I would remove preferred from the name

Reply:
A project has a specific goal/category, while a user would naturally have a list of categories there are interested in. So, the purpose of Group A could be primarily working on an AI/ML project, but while user1 might be interested in AI/ML, they might also having interests in frontend development.

That was my thought process. Do you want me to make it so that groups have more than one categories?

Reply:
That makes sense, I meant why don't you use the `Interest` model here instead of a String? Like how for `preferred_skills` you use the `Skill` model